site/content/en/docs/concepts/cluster_queue.md

```diff
@@ -497,17 +499,30 @@ spec:
 
 The fields above do the following:
 
-- `whenCanBorrow` determines whether a workload should stop finding a better assignment if it can get enough resource by borrowing in current ResourceFlavor. The possible values are:
-  - `Borrow` (default): ClusterQueue stops finding a better assignment.
-  - `TryNextFlavor`: ClusterQueue tries the next ResourceFlavor to see if the workload can get a better assignment.
-- `whenCanPreempt` determines whether a workload should try preemption in current ResourceFlavor before try the next one. The possible values are:
-  - `Preempt`: ClusterQueue stops trying preemption in current ResourceFlavor and starts from the next one if preempting failed.
-  - `TryNextFlavor` (default): ClusterQueue tries the next ResourceFlavor to see if the workload can fit in the ResourceFlavor.
-
-By default, the incoming workload stops trying the next flavor if the workload can get enough borrowed resources.
-And Kueue triggers preemption only after Kueue determines that the remaining ResourceFlavors can't fit the workload.
-
-Note that, whenever possible and when the configured policy allows it, Kueue avoids preemptions if it can fit a Workload by borrowing.
+- `whenCanBorrow` defines what should happen if a workload can get enough resource by
+borrowing in current ResourceFlavor. The possible values are:
+  - `Borrow` (default): workload stops looking for a better assignment.
+  - `TryNextFlavor`: workload tries the next ResourceFlavor.
+- `whenCanPreempt` defines what should happen if a workload can get enough resource by
+preempting in current ResourceFlavor. The possible values are:
+  - `Preempt`: workload stops looking for a better assignment.
+  - `TryNextFlavor` (default): workload tries the next ResourceFlavor.
+
+If during the search, the workload finds some ResourceFlavor in which it can fit
+without preemption or borrowing, such ResourceFlavor is immediately selected,
+regardless of the above configuration. Otherwise, out of the considered
+ResourceFlavors, Kueue selects a one that fits the workload using borrowing
+(without preemptions). If there is no such ResourceFlavor, Kueue selects a Flavor
+that uses preemption and is preferably not borrowing.
+
+By default Kueue avoids preemptions and prefers borrowing when assigning Flavors.
+Borrowing is not disruptive to other workloads but a
+workload that borrows risks being prempted (since it is using nominal quota
+from some other Cluster Queue). If you prefer to preempt rather than borrow when possible,
+you can enable the feature gate `FlavorFungibilityImplicitPreferenceDefault`, which
+changes the default preference as follows: If `whenCanBorrow = TryNextFlavor`
+it assumes that preemption is preferred over borrowing and otherwise it assumes
+that borrowing is preferred over preemption.
```

Comment:
Additionally, could you mention which Kueue version supports this FG like the following Note card?

```
AdmissionFairSharing is currently an alpha feature and is not enabled by default.

You can enable it by editing the AdmissionFairSharing feature gate. Check the [Installation](https://kueue.sigs.k8s.io/docs/installation/#change-the-feature-gates-configuration) guide for details on feature gate configuration.
```

https://kueue.sigs.k8s.io/docs/concepts/admission_fair_sharing/